v3/src/components/inspector-panel.tsx

```diff
@@ -72,20 +72,27 @@
   children: ReactNode
   Icon?: ReactNode
   title?: string
-  paletteTop?: number
   panelRect?: DOMRect
   buttonRect?: DOMRect
   setShowPalette: (palette: string | undefined) => void
 }
 
-export const InspectorPalette = ({children, Icon, title, paletteTop = 0,  panelRect, buttonRect,
+export const InspectorPalette = ({children, Icon, title, panelRect, buttonRect,
      setShowPalette}:IInspectorPalette) => {
   const panelTop = panelRect?.top || 0
   const buttonTop = buttonRect?.top || 0
   const [paletteWidth, setPaletteWidth] = useState(0)
   const paletteRef = useRef<HTMLDivElement>(null)
+  const pointerRef = useRef<HTMLDivElement>(null)
   const viewportEl = document.getElementsByClassName("free-tile-row")?.[0]
   const [inBounds, setInBounds] = useState(isWithinBounds(paletteWidth, panelRect))
+  const paletteHeight = paletteRef.current?.offsetHeight
+  const tempPaletteTop = paletteRef.current?.getBoundingClientRect().top
+  const pointerTop = buttonTop - panelTop - 5
+  const pointerHeight = pointerRef.current?.offsetHeight
+  const pointerMidpoint = pointerHeight ? pointerTop + pointerHeight/2 : 0
```

Comment:
I believe that if pointerHeight === 0, pointerMidpoint will be assigned the value of 0. Is that what you want?